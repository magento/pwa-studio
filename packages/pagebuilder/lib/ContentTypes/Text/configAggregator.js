import { getAdvanced } from '../../utils';

export default node => {
    return {
        content: node.innerHTML,
        ...getAdvanced(node)
    };
};
