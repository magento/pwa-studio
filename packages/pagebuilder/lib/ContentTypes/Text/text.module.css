.root {
    display: flex;
    flex-direction: column;
    word-wrap: break-word;
}
