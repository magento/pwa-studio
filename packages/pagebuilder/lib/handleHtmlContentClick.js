/**
 * Helper function for onClick() HTML Events
 *
 * @param {object} history history object
 * @param {function} history.push Pushes a new entry onto the history stack
 * @param {Event} event
 */
const handleHtmlContentClick = (history, event) => {
    const { code, target, type } = event;

    // Check if element is clicked or using accepted keyboard event
    const shouldIntercept =
        type === 'click' || code === 'Enter' || code === 'Space';

    // Intercept link clicks and check to see if the
    // destination is internal to avoid refreshing the page
    if (target.tagName === 'A' && shouldIntercept) {
        event.preventDefault();

        const eventOrigin = event.view.location.origin;

        const {
            origin: linkOrigin,
            pathname: path,
            target: tabTarget,
            href
        } = target;

        if (tabTarget && globalThis.open) {
            globalThis.open(href, '_blank');
        } else if (linkOrigin === eventOrigin) {
            history.push(path);
        } else {
            globalThis.location.assign(href);
        }
    }
};

export default handleHtmlContentClick;
