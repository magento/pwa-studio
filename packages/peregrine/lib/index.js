import * as RestApi from './RestApi';
import * as Util from './util';

// hooks
export { useEventListener } from './hooks/useEventListener';
export { useCarousel } from './hooks/useCarousel';
export { useDropdown } from './hooks/useDropdown';
export { usePagination } from './hooks/usePagination';
export { useRestApi } from './hooks/useRestApi';
export { useRestResponse } from './hooks/useRestResponse';
export { useScrollLock } from './hooks/useScrollLock';
export { useSearchParam } from './hooks/useSearchParam';
export { useSort } from './hooks/useSort';
export { useTypePolicies } from './hooks/useTypePolicies';

export {
    WindowSizeContextProvider,
    useWindowSize
} from './hooks/useWindowSize';
export { getToastId, useToasts, ToastContextProvider } from './Toasts';

// store
export { enhancer, reducers } from './store';

// components
export { default as ContainerChild } from './ContainerChild';
export { default as List, Items, Item } from './List';
export { default as Price } from './Price';
export { default as Router } from './Router';
export {
    default as PeregrineContextProvider
} from './PeregrineContextProvider';

// misc
export { RestApi };
export { Util };
export { default as createTestInstance } from './util/createTestInstance';
