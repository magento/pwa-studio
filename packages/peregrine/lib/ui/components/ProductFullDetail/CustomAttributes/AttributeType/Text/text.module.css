.label {
    composes: font-bold from global;
    composes: inline-block from global;
    composes: mr-2 from global;
}

.content {
    composes: inline-block from global;
}

.contentHtml {
    composes: inline-block from global;
}
