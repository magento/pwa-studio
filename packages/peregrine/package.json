{
  "name": "@magento/peregrine",
  "version": "12.1.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "The core runtime of Magento PWA",
  "main": "lib/index.js",
  "scripts": {
    "clean": " "
  },
  "repository": "github:magento/pwa-studio",
  "author": "Magento Commerce",
  "license": "(OSL-3.0 OR AFL-3.0)",
  "bugs": {
    "url": "https://github.com/magento/pwa-studio/issues"
  },
  "homepage": "https://github.com/magento/pwa-studio/tree/master/packages/peregrine#readme",
  "dependencies": {
    "fast-glob": "~3.2.4"
  },
  "devDependencies": {
    "@apollo/client": "~3.4.0",
    "@magento/eslint-config": "~1.5.0",
    "@testing-library/react-hooks": "~5.0.3",
    "intl": "~1.2.5",
    "intl-locales-supported": "~1.8.12",
    "react": "~17.0.1",
    "react-dom": "~17.0.1",
    "react-intl": "~5.20.0",
    "react-redux": "~7.2.2",
    "react-router-dom": "~5.2.0",
    "react-test-renderer": "~17.0.1",
    "redux": "~4.0.4",
    "redux-actions": "~2.6.5",
    "wait-for-expect": "~1.2.0"
  },
  "peerDependencies": {
    "@apollo/client": "~3.4.0",
    "@babel/runtime": "~7.15.3",
    "informed": "~3.29.0",
    "react": "~17.0.1",
    "react-dom": "~17.0.1",
    "react-intl": "~5.20.0",
    "react-redux": "~7.2.2",
    "react-router-dom": "~5.2.0",
    "redux": "~4.0.4",
    "redux-actions": "~2.6.5"
  },
  "engines": {
    "node": ">=14.x",
    "yarn": ">=1.12.0"
  },
  "pwa-studio": {
    "targets": {
      "declare": "./lib/targets/peregrine-declare",
      "intercept": "./lib/targets/peregrine-intercept"
    }
  },
  "module": "lib/index.js",
  "jsnext:main": "lib/index.js",
  "es2015": "lib/index.js",
  "sideEffects": false
}
