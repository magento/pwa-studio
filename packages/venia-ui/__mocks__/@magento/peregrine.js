import React from 'react';
import { BrowserPersistence } from './util/simplePersistence';

const peregrine = jest.requireActual('../../../peregrine/lib');

// re-exports
const RestApi = {
    Magento2: {
        request: jest.fn()
    }
};

const Util = { BrowserPersistence };

// hooks
const useEventListener = jest.fn(peregrine.useEventListener);
const useDropdown = jest.fn(peregrine.useDropdown);
const useSearchParam = jest.fn(peregrine.useSearchParam);
const useWindowSize = jest.fn(peregrine.useWindowSize);
const useToasts = jest.fn(peregrine.useToasts);
const useTypePolicies = jest.fn(peregrine.useTypePolicies);
// components

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
const Price = jest.fn().mockReturnValue(<div />);

module.exports = {
    ...peregrine,
    Price,
    RestApi,
    Util,
    useEventListener,
    useDropdown,
    useSearchParam,
    useWindowSize,
    useToasts,
    useTypePolicies
};
