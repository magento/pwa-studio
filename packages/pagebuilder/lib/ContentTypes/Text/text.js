import React from 'react';
import { arrayOf, shape, string } from 'prop-types';
import { useStyle } from '@magento/venia-ui/lib/classify';
import defaultClasses from './text.css';

const toHTML = str => ({ __html: str });

/**
 * Page Builder Text component.
 *
 * This component is part of the Page Builder / PWA integration. It can be consumed without Page Builder.
 *
 * @typedef Text
 * @kind functional component
 *
 * @param {props} props React component props
 *
 * @returns {React.Element} A React component that displays a Text content type which contains content.
 */
const Text = props => {
    const classes = useStyle(defaultClasses, props.classes);
    const {
        content,
        textAlign,
        border,
        borderColor,
        borderWidth,
        borderRadius,
        marginTop,
        marginRight,
        marginBottom,
        marginLeft,
        paddingTop,
        paddingRight,
        paddingBottom,
        paddingLeft,
        cssClasses = []
    } = props;

    const dynamicStyles = {
        textAlign,
        border,
        borderColor,
        borderWidth,
        borderRadius,
        marginTop,
        marginRight,
        marginBottom,
        marginLeft,
        paddingTop,
        paddingRight,
        paddingBottom,
        paddingLeft
    };

    return (
        <div
            style={dynamicStyles}
            className={[classes.root, ...cssClasses].join(' ')}
            dangerouslySetInnerHTML={toHTML(content)}
        />
    );
};

/**
 * Props for {@link Text}
 *
 * @typedef props
 *
 * @property {Object} classes An object containing the class names for the Text
 * @property {String} classes.root CSS class for the root text element
 * @property {String} content Content to be rendered within the content type
 * @property {String} textAlign Alignment of content within the text
 * @property {String} border CSS border property
 * @property {String} borderColor CSS border color property
 * @property {String} borderWidth CSS border width property
 * @property {String} borderRadius CSS border radius property
 * @property {String} marginTop CSS margin top property
 * @property {String} marginRight CSS margin right property
 * @property {String} marginBottom CSS margin bottom property
 * @property {String} marginLeft CSS margin left property
 * @property {String} paddingTop CSS padding top property
 * @property {String} paddingRight CSS padding right property
 * @property {String} paddingBottom CSS padding bottom property
 * @property {String} paddingLeft CSS padding left property
 * @property {Array} cssClasses List of CSS classes to be applied to the component
 */
Text.propTypes = {
    classes: shape({
        root: string
    }),
    content: string,
    textAlign: string,
    border: string,
    borderColor: string,
    borderWidth: string,
    borderRadius: string,
    marginTop: string,
    marginRight: string,
    marginBottom: string,
    marginLeft: string,
    paddingTop: string,
    paddingRight: string,
    paddingBottom: string,
    cssClasses: arrayOf(string)
};

export default Text;
