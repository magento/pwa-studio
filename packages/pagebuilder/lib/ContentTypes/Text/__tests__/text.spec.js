import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import Text from '../text';

jest.mock('@magento/venia-ui/lib/classify');

jest.mock('react-router-dom', () => {
    return {
        useHistory: jest.fn()
    };
});

jest.mock('../../../handleHtmlContentClick');
import handleHtmlContentClick from '../../../handleHtmlContentClick';

test('renders a Text component', () => {
    const textProps = {
        content: '<p>Test text component.</p>'
    };
    const component = createTestInstance(<Text {...textProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('renders a Text component with all props configured', () => {
    const textProps = {
        content: '<p>Another text component.</p>',
        textAlign: 'right',
        border: 'solid',
        borderColor: 'red',
        borderWidth: '10px',
        borderRadius: '15px',
        marginTop: '10px',
        marginRight: '10px',
        marginBottom: '10px',
        marginLeft: '10px',
        paddingTop: '10px',
        paddingRight: '10px',
        paddingBottom: '10px',
        paddingLeft: '10px',
        cssClasses: ['test-class']
    };
    const component = createTestInstance(<Text {...textProps} />);

    expect(component.toJSON()).toMatchSnapshot();
});

test('on click calls the HTML content click handler', () => {
    const textProps = {
        content: '<p>Test text component.</p>'
    };

    const mockHtmlContentClick = jest.fn();
    handleHtmlContentClick.mockImplementation(mockHtmlContentClick);

    const event = {
        target: {
            tagName: 'P'
        },
        preventDefault: jest.fn()
    };

    const component = createTestInstance(<Text {...textProps} />);

    const htmlElement = component.root.find(instance => {
        return instance.props.dangerouslySetInnerHTML;
    });

    htmlElement.props.onClick(event);

    expect(mockHtmlContentClick).toHaveBeenCalled();
});
