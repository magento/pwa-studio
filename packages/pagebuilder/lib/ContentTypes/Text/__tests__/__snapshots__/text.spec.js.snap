// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Text component 1`] = `
<div
  className="root"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Test text component.</p>",
    }
  }
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
  style={
    Object {
      "border": undefined,
      "borderColor": undefined,
      "borderRadius": undefined,
      "borderWidth": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "paddingBottom": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "textAlign": undefined,
    }
  }
/>
`;

exports[`renders a Text component with all props configured 1`] = `
<div
  className="root test-class"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Another text component.</p>",
    }
  }
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
  style={
    Object {
      "border": "solid",
      "borderColor": "red",
      "borderRadius": "15px",
      "borderWidth": "10px",
      "marginBottom": "10px",
      "marginLeft": "10px",
      "marginRight": "10px",
      "marginTop": "10px",
      "paddingBottom": "10px",
      "paddingLeft": "10px",
      "paddingRight": "10px",
      "paddingTop": "10px",
      "textAlign": "right",
    }
  }
/>
`;
